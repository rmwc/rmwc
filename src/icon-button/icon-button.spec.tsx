import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { IconButton } from './';

describe('', () => {
  it('renders with icon as string', () => {
    const { asFragment } = render(
      <IconButton icon="star" label="Rate this!" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with icon as path', () => {
    const { asFragment } = render(
      <IconButton icon="images/icons/twitter.png" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with icon as JSX', () => {
    const { asFragment } = render(
      <IconButton icon={<div />} label="Tweet it!" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders as toggle', () => {
    const { asFragment } = render(
      <IconButton onIcon="favorite" icon="favorite_border" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders controlled', () => {
    const { asFragment } = render(
      <IconButton
        checked={true}
        onClick={() => {}}
        onIcon="star"
        icon="star_border"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('handles onChange', async () => {
    let value = 0;
    render(
      <IconButton
        onChange={() => (value += 1)}
        onIcon="favorite"
        icon="favorite_border"
      />
    );

    userEvent.click(screen.getByRole('button'));

    await waitFor(() => expect(value).toEqual(1));
  });

  it('can have custom classnames', () => {
    const { container } = render(
      <IconButton icon="star" className={'my-custom-classname'} />
    );
    expect(container.firstChild).toHaveClass('my-custom-classname');
  });
});
