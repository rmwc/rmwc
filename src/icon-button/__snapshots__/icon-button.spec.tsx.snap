// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders as toggle 1`] = `
<DocumentFragment>
  <button
    aria-pressed="false"
    class="mdc-ripple-upgraded--unbounded mdc-icon-button"
    data-mdc-ripple-is-unbounded="true"
    role="button"
    tabindex="0"
  >
    <i
      class="rmwc-icon rmwc-icon--ligature material-icons mdc-icon-button__icon"
    >
      favorite_border
    </i>
    <i
      class="rmwc-icon rmwc-icon--ligature material-icons mdc-icon-button__icon mdc-icon-button__icon--on"
    >
      favorite
    </i>
  </button>
</DocumentFragment>
`;

exports[` renders controlled 1`] = `
<DocumentFragment>
  <button
    aria-pressed="true"
    class="mdc-ripple-upgraded--unbounded mdc-icon-button--on mdc-icon-button mdc-icon-button--on"
    data-mdc-ripple-is-unbounded="true"
    role="button"
    tabindex="0"
  >
    <i
      class="rmwc-icon rmwc-icon--ligature material-icons mdc-icon-button__icon"
    >
      star_border
    </i>
    <i
      class="rmwc-icon rmwc-icon--ligature material-icons mdc-icon-button__icon mdc-icon-button__icon--on"
    >
      star
    </i>
  </button>
</DocumentFragment>
`;

exports[` renders with icon as JSX 1`] = `
<DocumentFragment>
  <button
    aria-label="Tweet it!"
    class="rmwc-icon rmwc-icon--component material-icons mdc-ripple-upgraded--unbounded mdc-icon-button"
    data-mdc-ripple-is-unbounded="true"
    role="button"
    tabindex="0"
  >
    <div />
  </button>
</DocumentFragment>
`;

exports[` renders with icon as path 1`] = `
<DocumentFragment>
  <button
    class="rmwc-icon rmwc-icon--url material-icons mdc-ripple-upgraded--unbounded mdc-icon-button"
    data-mdc-ripple-is-unbounded="true"
    role="button"
    style="background-image: url(images/icons/twitter.png);"
    tabindex="0"
  />
</DocumentFragment>
`;

exports[` renders with icon as string 1`] = `
<DocumentFragment>
  <button
    aria-label="Rate this!"
    class="rmwc-icon rmwc-icon--ligature material-icons mdc-ripple-upgraded--unbounded mdc-icon-button"
    data-mdc-ripple-is-unbounded="true"
    role="button"
    tabindex="0"
  >
    star
  </button>
</DocumentFragment>
`;
